src/main/java/com/iota/iri/service/API.java

```diff
@@ -796,6 +797,12 @@ private synchronized AbstractResponse findTransactionStatement(final Map<String,
                 tag = padTag(tag);
                 tagsTransactions.addAll(TagViewModel.load(instance.tangle, new Hash(tag)).getHashes());
             }
+            if (tagsTransactions.isEmpty()) {
+                for (String tag : tags) {
+                    tag = padTag(tag);
+                    tagsTransactions.addAll(TagViewModel.loadObsolete(instance.tangle, new Hash(tag)).getHashes());
+                }
```

Comment:
It will be nice if the new block will be encapsulated in a method

Reply:
yeah, it would e nice :) , but not going to do that - not without rewriting all `findTransactionStatement`.